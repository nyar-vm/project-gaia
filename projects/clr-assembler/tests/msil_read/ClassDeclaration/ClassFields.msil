.class private auto ansi beforefieldinit
  Valkyrie.Translator.ClassAndStructure.ClassFields
    extends [System.Runtime]System.Object
{

  .field private int32 _field1

  .field family int32 _field2

  .field assembly int32 _field3

  .field public int32 _field4

  .field famorassem int32 _field5

  .field famandassem int32 _field6

  .field private initonly int32 '<_field11>k__BackingField'
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .custom instance void [System.Runtime]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggerBrowsableState)
      = (01 00 00 00 00 00 00 00 ) // ........
      // int32(0) // 0x00000000

  .field private int32 '<_field12>k__BackingField'
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .custom instance void [System.Runtime]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggerBrowsableState)
      = (01 00 00 00 00 00 00 00 ) // ........
      // int32(0) // 0x00000000

  .field private int32 '<_field13>k__BackingField'
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .custom instance void [System.Runtime]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [System.Runtime]System.Diagnostics.DebuggerBrowsableState)
      = (01 00 00 00 00 00 00 00 ) // ........
      // int32(0) // 0x00000000

  .field private static literal int32 _field21 = int32(100) // 0x00000064

  .field private static int32 _field22

  .field private initonly int32 _field23

  .method public hidebysig specialname instance int32
    get__field11() cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [12 27 - 12 31]
    IL_0000: ldarg.0      // this
    IL_0001: ldfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::'<_field11>k__BackingField'
    IL_0006: ret

  } // end of method ClassFields::get__field11

  .method public hidebysig specialname instance int32
    get__field12() cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [13 27 - 13 31]
    IL_0000: ldarg.0      // this
    IL_0001: ldfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::'<_field12>k__BackingField'
    IL_0006: ret

  } // end of method ClassFields::get__field12

  .method public hidebysig specialname instance void
    set__field12(
      int32 'value'
    ) cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [13 32 - 13 36]
    IL_0000: ldarg.0      // this
    IL_0001: ldarg.1      // 'value'
    IL_0002: stfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::'<_field12>k__BackingField'
    IL_0007: ret

  } // end of method ClassFields::set__field12

  .method public hidebysig specialname instance int32
    get__field13() cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [14 27 - 14 31]
    IL_0000: ldarg.0      // this
    IL_0001: ldfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::'<_field13>k__BackingField'
    IL_0006: ret

  } // end of method ClassFields::get__field13

  .method private hidebysig specialname instance void
    set__field13(
      int32 'value'
    ) cil managed
  {
    .custom instance void [System.Runtime]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor()
      = (01 00 00 00 )
    .maxstack 8

    // [14 32 - 14 44]
    IL_0000: ldarg.0      // this
    IL_0001: ldarg.1      // 'value'
    IL_0002: stfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::'<_field13>k__BackingField'
    IL_0007: ret

  } // end of method ClassFields::set__field13

  .method public hidebysig specialname rtspecialname instance void
    .ctor() cil managed
  {
    .maxstack 8

    // [18 5 - 18 33]
    IL_0000: ldarg.0      // this
    IL_0001: ldc.i4       300 // 0x0000012c
    IL_0006: stfld        int32 Valkyrie.Translator.ClassAndStructure.ClassFields::_field23
    IL_000b: ldarg.0      // this
    IL_000c: call         instance void [System.Runtime]System.Object::.ctor()
    IL_0011: nop
    IL_0012: ret

  } // end of method ClassFields::.ctor

  .method private hidebysig static specialname rtspecialname void
    .cctor() cil managed
  {
    .maxstack 8

    // [17 5 - 17 31]
    IL_0000: ldc.i4       200 // 0x000000c8
    IL_0005: stsfld       int32 Valkyrie.Translator.ClassAndStructure.ClassFields::_field22
    IL_000a: ret

  } // end of method ClassFields::.cctor

  .property instance int32 _field11()
  {
    .get instance int32 Valkyrie.Translator.ClassAndStructure.ClassFields::get__field11()
  } // end of property ClassFields::_field11

  .property instance int32 _field12()
  {
    .get instance int32 Valkyrie.Translator.ClassAndStructure.ClassFields::get__field12()
    .set instance void Valkyrie.Translator.ClassAndStructure.ClassFields::set__field12(int32)
  } // end of property ClassFields::_field12

  .property instance int32 _field13()
  {
    .get instance int32 Valkyrie.Translator.ClassAndStructure.ClassFields::get__field13()
    .set instance void Valkyrie.Translator.ClassAndStructure.ClassFields::set__field13(int32)
  } // end of property ClassFields::_field13
} // end of class Valkyrie.Translator.ClassAndStructure.ClassFields
