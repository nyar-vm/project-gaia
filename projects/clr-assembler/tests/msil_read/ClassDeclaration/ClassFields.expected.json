{
    "assembly_name": null,
    "extern_assemblies": [],
    "module_name": null,
    "class_name": "Valkyrie.Translator.ClassAndStructure.ClassFields",
    "class_modifiers": [
        "private",
        "auto",
        "ansi",
        "beforefieldinit"
    ],
    "extends": "[System.Runtime]System.Object",
    "method_count": 7,
    "method_names": [
        "get__field11",
        "get__field12",
        "set__field12",
        "get__field13",
        "set__field13",
        ".ctor",
        "unknown"
    ],
    "statement_count": 1
}